Libraries/LibWeb/CSS/CSSNumericType.cpp

```diff
@@ -145,16 +145,16 @@ Optional<CSSNumericType> CSSNumericType::added_to(CSSNumericType const& other) c
     // 3. If all the entries of type1 with non-zero values are contained in type2 with the same value, and vice-versa
     if (type2.contains_all_the_non_zero_entries_of_other_with_the_same_value(type1)
         && type1.contains_all_the_non_zero_entries_of_other_with_the_same_value(type2)) {
-        // Copy all of type1’s entries to finalType, and then copy all of type2’s entries to finalType that
-        // finalType doesn’t already contain. Set finalType’s percent hint to type1’s percent hint. Return finalType.
+        // Copy all of type1's entries to finalType, and then copy all of type2's entries to finalType that
+        // finalType doesn't already contain. Set finalType's percent hint to type1's percent hint. Return finalType.
         final_type.copy_all_entries_from(type1, SkipIfAlreadyPresent::No);
         final_type.copy_all_entries_from(type2, SkipIfAlreadyPresent::Yes);
         final_type.set_percent_hint(type1.percent_hint());
         return final_type;
     }
     //    If type1 and/or type2 contain "percent" with a non-zero value,
     //    and type1 and/or type2 contain a key other than "percent" with a non-zero value
-    if ((type1.exponent(BaseType::Percent) != 0 || type2.exponent(BaseType::Percent) != 0)
+    if ((type1.exponent(BaseType::Percent).has_value() || type2.exponent(BaseType::Percent).has_value())
```

Comment:
The spec says it needs to be a non-zero value, I think the old code was valid.

Reply:
Perhaps the language of the spec is a bit subtle here. Maybe by value, it means non-null. In which case this should still be checking for has_value, as well as for non-zero.

i.e

percent1.has_value() && percent1.value() != 0

Reply:
> percent1.has_value() && percent1.value() != 0

Isn't that exactly what `percent1 != 0` does?

Reply:
null is not equal to zero, so i dont believe so

Reply:
Oh ok, my bad, I'm just understanding what you're saying.